subprojects/logging/src/main/java/org/gradle/internal/logging/console/ProgressBar.java

```diff
@@ -60,6 +64,16 @@ public void moreProgress(int totalProgress) {
 
     public void update(boolean failing) {
         this.current++;
+        if (current > total) {
```

Comment:
🤔 I'm not convinced of this solution. It seems excessive that we start a thread for this. 

If we hit the condition `current > total`, we are going to start a thread for _each_ invocation of `update`.
This method will be called from `BuildStatusRendeded.onOutput`

https://github.com/gradle/gradle/blob/eb27e930e9120bff77261c49818111fcfda2bcdc/subprojects/logging/src/main/java/org/gradle/internal/logging/console/BuildStatusRenderer.java#L101-L108

I don't have an exact place where the events are created, but I'm afraid that we get an event for every build operation, and that would mean a _lot_ of threads created.

Depending on these factors, it might be better not to log anything. Looking at [`OutputEventListenerBackedLoggerContext`](https://github.com/gradle/gradle/blob/eb27e930e9120bff77261c49818111fcfda2bcdc/subprojects/logging/src/main/java/org/gradle/internal/logging/slf4j/OutputEventListenerBackedLoggerContext.java), there would be ways to extend the class a bit not to emit events when logging.

WDYT?

Reply:
I considered not logging it, but the problem with not logging anything is, that you then will not get reports about such incidents.
My intention was, that there are warning printed to increase the probability that users actually report such cases, that should always be bugs of either `total` not being set correctly, or `update` being called wrongly.

And usually, the case should not happen, as it is a bug if it happens.
So I thought if it actually happens, it would be ok to start threads there.

Also, I don't think it would be lots of threads, unless there is much discrepancy between the `current` and `total`.
`total` is the amount of actions in the build, so mainly the amount of tasks that are being executed. Some other things too, like artifact transformations done.
For example in #24370 which was the trigger for this fix, `total` was 1225 and there were 54 additional `update` calls. So yes, in this quite big build, there would have been 54 Threads to log this warning if that is what you call "lot". It of course is "a lot", but caused by a bug and if no bug is there, no threads will be created. :-)

If you are concerned about the amount of threads, how about a single-thread-executor that gets lazily initialized on first exccessive "update" call and then is reused? This way it would only be one additional thread.

Reply:
> I considered not logging it, but the problem with not logging anything is, that you then will not get reports about such incidents.

I agree with this, I also think not logging should be a last resort to fix this.

> like artifact transformations done.

This is as example I'm afraid of, as AGP for example relies on transformations heavily. If we have any oops there, we might create ridiculous amount of threads.

> If you are concerned about the amount of threads, how about a single-thread-executor that gets lazily initialized on first exccessive "update" call and then is reused? This way it would only be one additional thread.

That sounds much better! Can I ask you to implement it?

Reply:
Sure, no problem at all. Update pushed.